src/sparkx/Histogram.py

```diff
@@ -286,33 +302,39 @@ def remove_bin(self, index):
         ValueError
             If `index` is out of range.
         """
+        if self.number_of_bins_ is None:
```

Comment:
This checks would not be necessary if the instance checks in the constructor were still present.